org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/resourceloader/ResourceLoaderProviders.java

```diff
@@ -42,15 +44,39 @@ public Sorter getResourceSorter() {
 
 	/** Returns a loader with a parallelization degree of 2 to 4 (depending on how many processors are available). */
 	public static Provider<IResourceLoader> getParallelLoader() {
+		int nThreads = getNumberOfThreads();
+		return getParallelLoader(nThreads, 0);
+	}
+
+	/**
+	 * since 2.12
+	 */
+	public static Provider<IResourceLoader> getParallelLoader(long timeoutTime, TimeUnit timeoutUnit) {
+		int nThreads = getNumberOfThreads();
+		return getParallelLoader(nThreads, 0, timeoutTime, timeoutUnit);
+	}
+
+	/**
+	 * calculates the number of threads to be used for parallel resource loading
+	 * since 2.12
+	 */
+	public static int getNumberOfThreads() {
 		int nProcessors = Runtime.getRuntime().availableProcessors();
 		int nThreads = Math.max(2, Math.min(4, nProcessors));
-		return getParallelLoader(nThreads, 0);
+		return nThreads;
 	}
 
 	public static Provider<IResourceLoader> getParallelLoader(final int nrOfThreads) {
 		return getParallelLoader(nrOfThreads, 0);
 	}
 
+	/**
+	 * @since 2.12
+	 */
+	public static Provider<IResourceLoader> getParallelLoader(final int nrOfThreads, long timeoutTime, TimeUnit timeoutUnit) {
```

Comment:
no, just this one method is superfluous. You can use `getParallelLoader(final int nrOfThreads, final int bufferSize, long timeoutTime, TimeUnit timeoutUnit)`